department-of-veterans-affairs/caseflow
# Update existing virtual hearing logic to use new recipient tables

Stack 🥞  (2/4)
Follows: https://github.com/department-of-veterans-affairs/caseflow/pull/16511
Precedes: https://github.com/department-of-veterans-affairs/caseflow/pull/16520

### Description
This PR updates existing Virtual Hearing logic to account for the new `HearingEmailRecipient` table. 
- anytime a new `VirtualHearing` is created, instead of storing email information on the `VirtualHearing` model, store in respective `HearingEmailRecipient` table 
- anytime a `VirtualHearing` email information is updated, update the respective `HearingEmailRecipient` tables
- create accessors for the new email fields in `virtual_hearing.rb` so we can be backward compatible
- update `virtual_hearing_repository` logic to take new tables into account

### Testing Plan
1. Login as `BVASYELLOW` 
2. Test every aspect of virtual hearings:
     - conversion from hearing details
     - conversion back to original type from hearing details 
     - scheduling directly to virtual from case details 
     - scheduling directly to virtual from daily docket
     - email validations for all recipients
     - ensure email events display correctly
